tests/std/tests/Dev11_0863628_atomic_compare_exchange/test.cpp

```diff
@@ -553,6 +558,70 @@ void test_double_identical_results() {
 #endif // _HAS_CXX20
 }
 
+// Also test GH-4688 "<atomic>: atomic_ref<void*> and atomic<void*> lack difference_type"
+template <class, class = void>
+constexpr bool atomic_has_member_difference_type = false;
+template <class T>
+constexpr bool atomic_has_member_difference_type<T, void_t<typename atomic<T>::difference_type>> = true;
+
+STATIC_ASSERT(
+    atomic_has_member_difference_type<signed char>&& is_same_v<atomic<signed char>::difference_type, signed char>);

```

Comment:
There's no need to positively test `atomic_has_member_difference_type`, which greatly increases verbosity here. Mentioning `atomic<MEOW>::difference_type` will already verify its existence.